.buildkite/pipeline.yml

```diff
@@ -0,0 +1,105 @@
+env:
+  BUILDKITE_PLUGINS_ALWAYS_CLONE_FRESH: 1
+  # Uncomment this to get more logs during the electron-builder calls.
+  # Useful for debugging, but very noisy
+  # DEBUG: electron-builder
+
+steps:
+  - label: Lint
+    key: lint
+    env:
+      NODE_ENV: test
+    plugins:
+      - $NVM_PLUGIN
+    command: |
+      .buildkite/commands/install_node_dependencies.sh
+      echo "--- :eslint: Lint"
+      make lint
+
+  - label: Test
+    key: test
+    env:
+      NODE_ENV: test
+    plugins:
+      - $NVM_PLUGIN
+    command: |
+      .buildkite/commands/install_node_dependencies.sh
+      echo "--- :jest: Test"
+      npm test
+
+  # Notice that we build the app in each platform because it takes ~1-2 minutes
+  # to do so which is comparable to a dedicated build step, archiving the
+  # artifacts, uploading them, then downloading them in each package step.
+  #
+  # Also notice that we package for the different platforms on every build.
+  # It's up to the elector-builder configuration to decided whether to upload
+  # the artifacts to a GitHub release.
+  #
+  # See:
+  # - Makefile PUBLISH value
+  # - https://www.electron.build/configuration/publish
+
+  - label: Package on macOS
+    key: package-macos
+    agents:
+      queue: mac
+    env:
+      IMAGE_ID: $IMAGE_ID
+      CSC_FOR_PULL_REQUEST: true
+    plugins:
+      - $CI_TOOLKIT_PLUGIN
+      - $NVM_PLUGIN
+    command: |
+      .buildkite/commands/install_node_dependencies.sh
+      echo "--- Fetch code signing identity"
+      install_gems
+      bundle exec fastlane configure_code_signing
+      echo "--- Decrypt secrets"
+      # This comes from 'make decrypt_conf' which we cannot call because it's interactive
+      openssl aes-256-cbc -d -in ./resources/secrets/config.json.enc -out ./config-local.json -pbkdf2 -k $SECRETS_ENCRYPTION_KEY
+      bundle exec fastlane run configure_apply
+      echo "--- Build"
+      make build
+      echo "--- Package"
+      make package-osx SKIP_BUILD=true
```

Comment:
I considered running `make build` standalone and then download the artifacts, but the overhead at buildtime is minimal compared to the rest of the step.